src/components/ChatAutoComplete/__tests__/ChatAutocomplete.test.js

```diff
@@ -0,0 +1,180 @@
+  const typeText = (text) => {
+    fireEvent.change(textarea, {
+      target: {
+        value: text,
+        selectionEnd: text.length,
+      },
+    });
+  };
+  return { ...renderResult, typeText, textarea };
+};
+
+describe('ChatAutoComplete', () => {
+  it('should call onChange with the change event when you type in the input', async () => {
+    const onChange = jest.fn();
+    const { typeText } = await renderComponent({ onChange });
+    typeText('something');
+    expect(onChange).toHaveBeenCalledWith(
+      expect.objectContaining({
+        target: expect.objectContaining({
+          value: 'something',
+        }),
+      }),
+    );
+  });
+
+  it('should pass the placeholder prop into the textarea', async () => {
+    const placeholder = 'something';
+    const { getByPlaceholderText } = await renderComponent({ placeholder });
+
+    expect(getByPlaceholderText(placeholder)).toBeInTheDocument();
+  });
+
+  it('should pass the disabled prop to the textarea', async () => {
+    const { textarea } = await renderComponent({ disabled: true });
+
+    expect(textarea).toBeDisabled();
+  });
+
+  it('should let you select emojis when you type :<emoji>', async () => {
+    const emojiAutocompleteText = ':smile';
+    const { typeText, findByText, textarea } = await renderComponent();
+    typeText(emojiAutocompleteText);
+    const emoji = await findByText('😄');
+
+    expect(emoji).toBeInTheDocument();
+
+    fireEvent.click(emoji);
+
+    expect(textarea.value).toContain('😄');
+  });
+
+  it('should let you select users when you type @<username>', async () => {
+    const onSelectItem = jest.fn();
+    const userAutocompleteText = `@${user.name}`;
+    const { typeText, findByText } = await renderComponent({ onSelectItem });
+    typeText(userAutocompleteText);
+    const userText = await findByText(user.name);
+
+    expect(userText).toBeInTheDocument();
+
+    fireEvent.click(userText);
+
+    expect(onSelectItem).toHaveBeenCalledWith(
+      expect.objectContaining({
+        id: user.id,
+      }),
+    );
+  });
+
+  it('should let you select users when you type @<userid>', async () => {
+    const onSelectItem = jest.fn();
+    const userAutocompleteText = `@${user.id}`;
+    const { typeText, findByText } = await renderComponent({ onSelectItem });
+    typeText(userAutocompleteText);
+    const userText = await findByText(user.name);
+
+    expect(userText).toBeInTheDocument();
+
+    fireEvent.click(userText);
+
+    expect(onSelectItem).toHaveBeenCalledWith(
+      expect.objectContaining({
+        id: user.id,
+      }),
+    );
+  });
+
+  it('should use the queryMembers API for mentions if a channel has many members', async () => {
+    const users = Array(100).fill().map(generateUser);
+    const members = users.map((u) => generateMember({ user: u }));
+    const messages = [generateMessage({ user: users[0] })];
+    const queryMembers = jest.fn(() => Promise.resolve({ members }));
```

Comment:
I will suggest, lets stick to `useMockedApis` instead of mocking functions manually on js client classes.

Reply:
I changed the tests to do this. It did mean I had to use the `Chat` and `Channel` component to provide the right context, and I had to write a little helper to set the active channel. It seems that this changes some state in the stream-chat client/channel that is needed for everything to work properly.

Can you please check if what I did makes sense to you? I feel like there might be an easier way to do this, but I'm not totally sure if that's actually the case.